File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int vote = 0;
        int candidate = 0;
        for(int num: nums) {
            if (vote == 0) candidate = num;
            if (candidate == num) vote++;
            else vote--;
        }
        return candidate;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        candidate = nums[0]
        vote = 1
        for j in range(1, len(nums)):
            if nums[j] == candidate:
                vote += 1
            else: 
                vote -= 1
                if vote == -1:
                    candidate = nums[j]
                    vote = 1
        return candidate